src/hotspot/share/opto/library_call.cpp

```diff
@@ -57,6 +57,10 @@
 #include "utilities/macros.hpp"
 #include "utilities/powerOfTwo.hpp"
 
+#ifdef JFR_HAVE_INTRINSICS
+#include "jfr/recorder/checkpoint/types/traceid/jfrTraceIdEpoch.hpp"
```

Comment:
Can you move all entry points you need to expose to "jfr/jfr.hpp" and include that instead? I prefer to avoid exposing a lot of the internal impl. details if possible. You can then put move the SharedRuntime::trace_id_load_barrier() routines, into jfr/jfr.cpp and avoid touching the runtime/sharedRuntime.hpp | .cpp. Thanks.

Reply:
removed.